components/cloud/azure/src/kms.rs

```diff
@@ -0,0 +1,337 @@
+                    azure_cfg.client_id.clone(),
+                    certificate.clone(),
+                    "".to_owned(),
+                ),
+                ClientCertificateCredentialExt::new(
+                    azure_cfg.tenant_id.clone(),
+                    azure_cfg.client_id,
+                    certificate,
+                    "".to_owned(),
+                ),
+            );
+            Self::new_with_credentials(config, keyvault_credential, hsm_credential)
+        } else if let Some(certificate_path) = azure_cfg.client_certificate_path.clone() {
+            // Certificate recorded in a file to accessing KeyVault.
+            let (keyvault_credential, hsm_credential) = (
+                ClientCertificateCredentialExt::build(
+                    azure_cfg.tenant_id.clone(),
+                    azure_cfg.client_id.clone(),
+                    certificate_path.clone(),
+                )
+                .map_err(|e| CloudError::Other(e))?,
+                ClientCertificateCredentialExt::build(
+                    azure_cfg.tenant_id.clone(),
+                    azure_cfg.client_id,
+                    certificate_path,
+                )
+                .map_err(|e| CloudError::Other(e))?,
+            );
+            Self::new_with_credentials(config, keyvault_credential, hsm_credential)
+        } else if let Some(client_secret) = azure_cfg.client_secret.clone() {
+            // Client secret to access KeyVault.
+            let (keyvault_credential, hsm_credential) = (
+                ClientSecretCredential::new(
+                    new_http_client(),
+                    azure_cfg.tenant_id.clone(),
+                    azure_cfg.client_id.clone(),
+                    client_secret.clone(),
+                    TokenCredentialOptions::default(),
+                ),
+                ClientSecretCredential::new(
+                    new_http_client(),
+                    azure_cfg.tenant_id.clone(),
+                    azure_cfg.client_id,
+                    client_secret,
+                    TokenCredentialOptions::default(),
+                ),
+            );
+            Self::new_with_credentials(config, keyvault_credential, hsm_credential)
+        } else {
+            Err(CloudError::KmsError(KmsError::Other(box_err!(
+                "invalid configurations for Azure KMS"
+            ))))
+        }
+    }
+}
+
+#[async_trait]
+impl KmsProvider for AzureKms {
+    fn name(&self) -> &str {
+        ENCRYPTION_VENDOR_NAME_AZURE_KMS
+    }
+
+    // On decrypt failure, the rule is to return WrongMasterKey error in case it is
+    // possible that a wrong master key has been used, or other error
+    // otherwise.
+    async fn decrypt_data_key(&self, data_key: &EncryptedKey) -> Result<Vec<u8>> {
+        let decrypt_params = DecryptParameters {
+            ciphertext: data_key.clone().into_inner(),
+            decrypt_parameters_encryption: CryptographParamtersEncryption::Rsa(
```

Comment:
Could you add some comment about why we choose `EncryptionAlgorithm::RsaOaep256`?